File: caesar.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<cs50.h>
#include<ctype.h>
int main(int argc,string argv[]){
    if(argc!=2){
        printf("please provide proper arguments");
        return 1;
    }
    char cipher_text[50];
    int i=0;
    int j=0;
    int character;
    int inc=atoi(argv[1]);
  printf("plaintext: ");
    string plain_text=get_string();
    if(plain_text!=NULL){
        while(plain_text[i]!='\0'){
            if(isalpha(plain_text[i])){
               
                if(isupper(plain_text[i])){
                  character=(plain_text[i]-65+inc)%26+65; 
                  cipher_text[j]=character;
                }
                else {
                    character=(plain_text[i]-97+inc)%26+97; 
                  cipher_text[j]=character;
                }
            }
            else{
                cipher_text[j]=plain_text[i];
            }
            i++;
            j++;
        }
    }
    cipher_text[j]='\0';
    printf("ciphertext: %s\n",cipher_text);
    return 0;
}

File: caesar.py
import sys
if len(sys.argv)!=2:
    print("please provide proper arguments")
    exit(1)
i=0
j=0
inc=int(sys.argv[1])
plain_text=input("plaintext:")
cipher_text=plain_text
if plain_text!=None:
    while True:
        if (i==len(plain_text)):
            break
        if plain_text[i].isalpha():
            if plain_text[i].isupper():
                character=(ord(plain_text[i])-65+inc)%26+65
                
                plain_text=plain_text.replace(plain_text[i],chr(character),1)
            else:
                character=(ord(plain_text[i])-97+inc)%26+97
                
                plain_text=plain_text.replace(plain_text[i],chr(character),1)
        
        i=i+1
        
print("ciphertext: {}".format(plain_text))
exit(0)
